### PR TITLE
Allow generating large strings in benchmarks

### PR DESCRIPTION
**Pull request by davidwendt**
## Description
Updates the benchmark utility `create_random_utf8_string_column` to support large strings.
Replaces the hardcoded `size_type` offsets with the offsetalator and related utilities.

Reference #16948 

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/benchmarks/common/generate_input.cu [hunk 1]
```diff
@@ -580,28 +585,24 @@ std::unique_ptr<cudf::column> create_random_utf8_string_column(data_profile cons
...
                      num_rows,
                      string_generator{chars.data(), engine});
+
   auto [result_bitmask, null_count] =
-    cudf::detail::valid_if(null_mask.begin(),
-                           null_mask.end() - 1,
-                           thrust::identity<bool>{},
-                           cudf::get_default_stream(),
-                           cudf::get_current_device_resource_ref());
+    profile.get_null_probability().has_value()
```
Comment (mhaseeb123) [inline cpp/benchmarks/common/generate_input.cu:599]: Nice optimization here!

File: cpp/benchmarks/common/generate_input.cu [hunk 2]
```diff
@@ -580,28 +585,24 @@ std::unique_ptr<cudf::column> create_random_utf8_string_column(data_profile cons
   auto valid_lengths = thrust::make_transform_iterator(
     thrust::make_zip_iterator(thrust::make_tuple(lengths.begin(), null_mask.begin())),
     valid_or_zero{});
-  rmm::device_uvector<cudf::size_type> offsets(num_rows + 1, cudf::get_default_stream());
-  thrust::exclusive_scan(
-    thrust::device, valid_lengths, valid_lengths + lengths.size(), offsets.begin());
-  // offsets are ready.
-  auto chars_length = *thrust::device_pointer_cast(offsets.end() - 1);
+  auto [offsets, chars_length] = cudf::strings::detail::make_offsets_child_column(
```
Comment (mhaseeb123) [inline cpp/benchmarks/common/generate_input.cu:588]: Minor: Let's add a couple comments here would help if you think.

**Review by mhaseeb123 — APPROVED** [review]
A couple minor comments. LGTM otherwise!

File: cpp/benchmarks/common/generate_input.cu (hunk 1 above)
Comment (mythrocks) [reply cpp/benchmarks/common/generate_input.cu:601]: Yep, that made sense to do.

**Review by mythrocks — APPROVED** [review]

**Pull request merged**